### PR TITLE
Major generator/spec/Travis improvements

### PR DESCRIPTION
**Pull request by randomcamel**

### CODE REVIEW

File: bin/generate_driver [hunk 1]
```diff
@@ -118,6 +118,16 @@ EOS
 end
 
 # -----------------------------------------
+# machine resource basics.
```
Comment (tyler-ball) [inline bin/generate_driver]: What does this stuff mean?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,38 +1,16 @@
...
-# a separate helper file that requires the additional dependencies and performs
-# the additional setup, and require it from the spec files that actually need
-# it.
-#
-# The `.rspec` file also contains a few flags that are not defaults but that
-# users commonly want.
-#
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+require 'simplecov'
+SimpleCov.start do
```
Comment (tyler-ball) [inline spec/spec_helper.rb]: What are we covering in here?

File: bin/generate_driver (hunk 1 above)
Comment (randomcamel) [inline bin/generate_driver]: That's a somewhat WIP thing where I was about to start experimenting with running unit specs on the generated driver. I can remove it for this PR, though probably no one should be looking at this file anyway.

File: spec/spec_helper.rb (hunk 2 above)
Comment (randomcamel) [inline spec/spec_helper.rb]: Straight-up `chef-provisioning`.

Comment (tyler-ball) [inline spec/spec_helper.rb]: But it looks to me like everything between the `do`/`end` blocks is commented out - does it still capture coverage for the specs?

**Comment by tyler-ball** [comment]
:+1:

File: spec/spec_helper.rb (hunk 2 above)
Comment (randomcamel) [inline spec/spec_helper.rb]: Yep, the block just allows configuration frobbing, the `start` is the necessary secret sauce to run it.

**Pull request merged**